### PR TITLE
feat: include log messages for rsvp changes

### PR DESCRIPTION
**Pull request by IanWearsHat**
Also added unit test for exception case in `/rsvp`

### CODE REVIEW

File: apps/api/tests/test_user.py [hunk 1]
```diff
@@ -116,3 +116,19 @@ def test_user_with_status_attending_un_rsvp_changes_status_to_void(
...
     assert res.status_code == 303
+
+@patch("services.mongodb_handler.update_one", autospec=True)
+@patch("services.mongodb_handler.retrieve_one", autospec=True)
+def test_user_with_status_reviewed_un_rsvp_returns_403(
+    mock_mongodb_handler_retrieve_one: AsyncMock,
+    mock_mongodb_handler_update_one: AsyncMock,
+) -> None:
+    """Test user with REVIEWED status has no status change after RSVP."""
+    mock_mongodb_handler_retrieve_one.return_value = {"status": Status.REVIEWED}
```
Comment (taesungh) [inline apps/api/tests/test_user.py:128]: Suggestion: a more appropriate test would be with `Status.ACCEPTED` to emphasize that users cannot RSVP without submitting the waiver.

Comment (IanWearsHat) [reply apps/api/tests/test_user.py:128]: `ACCEPTED` is currently not a status in `Status` for `user_record.py`. Should I add it?

Comment (taesungh) [reply apps/api/tests/test_user.py:128]: Sorry I meant `Decision.ACCEPTED`

**Review by taesungh — APPROVED** [review]
Thanks for adding this is as well as the unit test.

**Review by samderanova — APPROVED** [review]
Thanks for working on this.

**Pull request merged**